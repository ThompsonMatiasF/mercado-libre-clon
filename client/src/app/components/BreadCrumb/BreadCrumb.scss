.breadcrumb-item + .breadcrumb-item::before {
  content: ">" !important;
  font-weight: normal;
}

.breadcrumb li{
  padding: 0;
}

.breadcrumb li:last-child {
  font-weight: bolder;
}

ol.breadcrumb {
  padding: 16px;
  margin: 0;
  font-size: 14px;
  line-height: 14px;
}

