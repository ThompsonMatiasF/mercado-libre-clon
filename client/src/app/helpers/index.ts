export { default as mocks } from './mocks'
