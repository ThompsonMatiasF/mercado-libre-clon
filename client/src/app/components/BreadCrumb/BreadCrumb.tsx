import React, { useContext } from 'react';
import { Breadcrumb } from 'react-bootstrap';
import './BreadCrumb.scss';
import { AppContext } from 'app/store';

export const BraedCrumb = () => {
  const { state: { breadcrumb: { categories } } } = useContext(AppContext);

  return (
    <Breadcrumb>
      { categories
      && categories.map((category: string) => <Breadcrumb.Item key={Math.random()} active>{ category }</Breadcrumb.Item>)}
    </Breadcrumb>
  );
};

export default BraedCrumb;
