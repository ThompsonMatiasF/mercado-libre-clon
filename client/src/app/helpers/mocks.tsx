import React, { FC } from 'react';
import { AppContext } from 'app/store';

interface MockContextProps {
  context: any
}

export const MockContext: FC<MockContextProps> = ({ context, children })  => (
  <AppContext.Provider value={ context }>{ children }</AppContext.Provider>
)

export const mockCategories = ['Celulares y Teléfonos', 'Celulares y Smartphones']

const mockPrice = {
  currency: 'ARS',
  amount: 100,
  decimals: 0,
}

export const mockItem = {
  id: 'MLA844506827',
  title: 'Iphone X 128gb',
  price: mockPrice,
  picture: 'https://domain.com/image.png',
  condition: 'new',
  free_shipping: true,
  address: 'Capital Federal',
}

export const mockItemDetail = {
  ...mockItem,
  categories: mockCategories,
  sold_quantity: 2,
  description: 'Iphone X 128g, nuevo, en caja cerrada con garantía oficial',
}

export const mockItems = Array(5).fill(mockItem)

export const mockSearchResult = {
  categories: mockCategories,
  items: mockItems,
}

export default {
  MockContext,
  mockItems,
  mockItem,
  mockItemDetail,
  mockSearchResult,
}

