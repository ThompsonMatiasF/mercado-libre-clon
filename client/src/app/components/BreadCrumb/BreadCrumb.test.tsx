import React from 'react';
import { render } from '@testing-library/react'
import BreadCrumb from './BreadCrumb'
import { MockContext, mockCategories } from 'app/helpers/mocks'

const context = {
  state: {
    breadcrumb: {
      categories: mockCategories,
    },
  },
}

describe('BreadCrumb.tsx', () => {
  it('should match snapshot', () => {
    const { container } = render(<MockContext context={context}><BreadCrumb /></MockContext>)

    expect(container).toMatchSnapshot()
  })
})
