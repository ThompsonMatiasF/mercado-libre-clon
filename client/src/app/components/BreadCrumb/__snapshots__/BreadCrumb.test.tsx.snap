// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb.tsx should match snapshot 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
  >
    <ol
      class="breadcrumb"
    >
      <li
        aria-current="page"
        class="breadcrumb-item active"
      >
        Celulares y Tel√©fonos
      </li>
      <li
        aria-current="page"
        class="breadcrumb-item active"
      >
        Celulares y Smartphones
      </li>
    </ol>
  </nav>
</div>
`;
